packages/packagers/js/src/JSPackager.js

```diff
@@ -18,29 +18,41 @@ export default new Packager({
 
     let assets = '';
     let i = 0;
-    bundle.assetGraph.traverseAssets(asset => {
-      let deps = {};
+    let first = true;
+    bundle.assetGraph.traverseAssetsWithReferences(({type, asset}) => {
+      if (type === 'asset_reference' && asset.type === 'js') {
+        // if this is a reference to another javascript asset, we should not include
+        // either its output or a stub, as its contents should already be loaded
+        return;
+      }
 
-      let dependencies = bundle.assetGraph.getDependencies(asset);
-      for (let dep of dependencies) {
-        let resolved = bundle.assetGraph.getDependencyResolution(dep);
-        if (resolved) {
-          deps[dep.moduleSpecifier] = resolved.id;
+      let wrapped = first ? '' : ',';
+      if (type === 'asset_reference') {
+        wrapped += JSON.stringify(asset.id) + ':[0,{}]';
```

Comment:
Instead of writing in a no-op function each time, pass `0` as a sentinel indicating this is a stub. Alternatives:

* declare a `noop` function and reference it here. This will require polluting `globalThis` or wrapping the prelude and this code in an iifee. This means the prelude doesn't need to change either, and keeps the shapes of these tuples predictable.
* passing `null` or `undefined` rather than `0`, which adds more bytes to the output source text and is somewhat more reasonable.

I decided to go with `0`, at the cost of complicating the shapes of these tuples, to save bytes and minimize polluting `globalThis`. Open to suggestions.

Reply:
I wouldn't worry too much about optimizing it. For prod builds, the tree shaking implementation will take care of most of this.

Reply:
How would it? 

Another thought: If these were `null` rather than `0` I could imagine terser creating a common reference, which would probably make this pretty negligible.

Reply:
Well with the tree shaker, references to modules that don't exist are just replaced with `{}`. No need for a function at all.